core/templates/dev/head/domain/exploration/FallbackObjectFactory.js

```diff
@@ -0,0 +1,38 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new frontend instances of Fallback
+ * domain objects.
+ */
+
+oppia.factory('FallbackObjectFactory', [function() {
+  var Fallback = function(triggerBackendDict, outcomeBackendDict) {
+    this.trigger = triggerBackendDict;
+    this.outcome = outcomeBackendDict;
+  };
+
+  Fallback.prototype.toBackendDict = function() {
+    return {
+      trigger: this.trigger,
+      outcome: this.outcome
+    };
+  };
+
+  Fallback.create = function(fallbackDict) {
```

Comment:
True--I realized the renamings I did in AnswerGroupObjectFactory aren't entirely accurate as sometimes its being created from front-end entities. So for AnswerGroupObjectFactory we could have a separate createFromBackendDict method (and rename the other .create methods) and also a separate createNew method that accepts outcome, trigger and correct as parameters. Is that along the lines of what you were thinking?

I noticed that 'false' is always being passed into AnswerGroupObjectFactory.create inside InteractionObjectFactory but why is it not answerGroupDict.correct?